pac4j-saml/src/main/java/org/pac4j/saml/metadata/SAML2MetadataGenerator.java

```diff
@@ -190,7 +192,7 @@ protected final SPSSODescriptor buildSPSSODescriptor() {
 
         int index = 0;
         spDescriptor.getAssertionConsumerServices()
-            .add(getAssertionConsumerService(SAMLConstants.SAML2_POST_BINDING_URI, index++, this.defaultACSIndex == index));
+            .add(getAssertionConsumerService(responseBindingType, index++, this.defaultACSIndex == index));
```

Comment:
I'm wondering if we shouldn't add both bindings (POST and artifact)?

Reply:
That should be possible, but it would require detecting the binding used or registering different services (urls). You can detect the artifact binding by the `SAMLart` parameter (query or post).

Reply:
I may be missing the point, but why not add the two bindings by default?

Reply:
In the metadata that would be possible, but then you'd have to detect the type of the binding used by the IdP on an incoming request. At the moment `SAML2Client` uses a different `SAML2MessageReceiver` when configured for the artifact binding (see `initSAMLProfileHandler`). I guess it should be possible to write a `SAML2MessageReceiver` that wraps both receivers and detects the binding being used.

When you provide both bindings, it's up to the IdP to make the choice. This might cause issues as well, because the artifact binding might require some additional configuration. If a client is not configured correctly to use the artifact binding, it will break when upgrading pac4j and then there would be no way to disable it.

Another problem might be ill-configured IdPs, that do not understand the metadata, but I can't say if that's a valid concern or not.

Rethinking this: maybe it's a good idea to make the `responseBindingType` a list and call it `responseBindingTypes`.

Reply:
OK. So let's keep one binding.

Reply:
Yes, as an IdP is only going to use one binding anyway, I guess it doesn't make much sense to support both simultaneously.